bundle/bundler/src/lib.rs

```diff
@@ -0,0 +1,220 @@
+use std::collections::BTreeMap;
+use std::path::{Path, PathBuf};
+
+use async_std::channel::bounded;
+use async_std::task;
+use async_std::task::block_on;
+
+use anyhow::Context;
+use anyhow::Result;
+use cid::multihash::Code;
+use cid::Cid;
+use fvm_shared::actor;
+use fvm_shared::actor::builtin::Manifest;
+use fvm_shared::blockstore::{Block, Blockstore, MemoryBlockstore};
+use fvm_shared::encoding::DAG_CBOR;
+use ipld_car::CarHeader;
+
+const IPLD_RAW: u64 = 0x55;
+
+/// A library to bundle the Wasm bytecode of builtin actors into a CAR file.
+///
+/// The single root CID of the CAR file points to an CBOR-encoded IPLD
+/// Map<Cid, i32> where i32 is to be interpreted as an
+/// fvm_shared::actor::builtin::Type enum value.
+pub struct Bundler {
+    /// Staging blockstore.
+    blockstore: MemoryBlockstore,
+    /// Tracks the mapping of actors to Cids. Inverted when writing. Allows overriding.
+    added: BTreeMap<fvm_shared::actor::builtin::Type, Cid>,
+    /// Path of the output bundle.
+    bundle_dst: PathBuf,
+}
+
+impl Bundler {
+    pub fn new<P>(bundle_dst: P) -> Bundler
+    where
+        P: AsRef<Path>,
+    {
+        Bundler {
+            bundle_dst: bundle_dst.as_ref().to_owned(),
+            blockstore: Default::default(),
+            added: Default::default(),
+        }
+    }
+
+    /// Adds bytecode from a byte slice.
+    pub fn add_from_bytes(
+        &mut self,
+        actor_type: actor::builtin::Type,
+        forced_cid: Option<Cid>,
```

Comment:
Ultra-nit: use `Option<&Cid>`. `Cid` is a pretty big type, so writing it this way will copy, like, 80 bytes or something.